src/engine/ruleset/decoders/sonicwall/firewall-general.yml

```diff
@@ -0,0 +1,77 @@
+---
+name: decoder/sonicwall-firewall-general/0
+
+metadata:
+  module: firewall
+  title: Firewall logs decoder
+  description: Decoder for Linux firewall logs
+  author:
+    name: Wazuh Inc. info@wazuh.com
+    date: 2023-01-12
+  references:
+    - https://www.sonicwall.com/es-mx/support/technical-documentation
+
+sources:
+  - decoder/syslog/0
+  - decoder/queue-localfile/0
+
+check:
+  - event.original: +r_match/id\=.*?sn\=.*?time\=.*?fw
+
+parse:
+  logpar:
+    # Jan  3 13:45:36 192.168.5.1 id=firewall sn=000SERIAL time="2007-01-03 14:48:06" fw=89.160.20.156 pri=6 c=262144 m=98 msg="Connection Opened" n=23419 src=2.2.2.2:36701:WAN dst=89.160.20.156:50000:WAN proto=tcp/50000
+    - event.original: >-
+        <~log.timestamp/date/%b %d %T> <host.ip> <~log.message>
+
+normalize:
+  - check:
+      - ~log.message: +ef_exists
+    logpar:
+      - ~log.message: <~tmp/kv/=/ /"/'>
```

Comment:
In this case `~log.message` must exist, if not then the `logpar` expression wouldn't succeed and the `normalize` stage wouldn't be executed, so this is redundant. On the other hand, you could resolve the `kv` operation directly on the `logpar` expression as follows:
```yml
- event.original: >-
        <~log.timestamp/date/%b %d %T> <host.ip> <~tmp/kv/=/ /"/'>
```

Additionally, `logpar` is not intended to be used in the `normalize` stage. Is this even working? Because as far as I know, it shouldn't.